Source/SwiftLintFramework/Rules/IdentifierNameRule.swift

```diff
@@ -70,7 +72,9 @@ public struct IdentifierNameRule: ASTRule, ConfigurationProviderRule {
                 }
             }
 
-            if kind != .varStatic && name.isViolatingCase && !name.isOperator {
+            let requiresCaseCheck = !configuration.ignoresStartWithLowercase || isFunction
```

Comment:
It made sense to me not to allow this option for functions, I thought there would be more cases where someone would want to disable this, in order to migrate code with enum cases that start with uppercase for example, than cases where someone would have functions starting with uppercase. Nonetheless, I wouldn't mind changing this if you think it'd be better 🙂